### PR TITLE
Start-up and loading improvements

### LINKED ISSUE #216: Issues running Sia-UI
When Running Sia-UI it asked me for siad to be running, once I select the siad file under the resources/Sia/app/siad it returns the following:

```
(node) warning: possible EventEmitter memory leak detected. 11 did-finish-load listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at EventEmitter.addListener (events.js:239:17)
    at signal (/home/user/Downloads/Sia/resources/app/js/mainjs/initSiad.js:22:6)
    at /home/user/Downloads/Sia/resources/app/js/mainjs/initSiad.js:44:3
    at /home/user/Downloads/Sia/resources/app/node_modules/sia.js/js/sia.js:98:9
    at /home/user/Downloads/Sia/resources/app/node_modules/sia.js/js/sia.js:83:9
    at Request._callback (/home/user/Downloads/Sia/resources/app/node_modules/sia.js/js/sia.js:69:9)
    at Request.self.callback (/home/user/Downloads/Sia/resources/app/node_modules/request/request.js:199:22)
    at emitTwo (events.js:87:13)
    at Request.emit (events.js:172:7)
    at Request.<anonymous> (/home/user/Downloads/Sia/resources/app/node_modules/request/request.js:1036:10)
```

I only get a white screen. Please help.

### PR DESCRIPTION
**Pull request by avahowell**
This pull request improves the functionality of Sia-UI's loading screen and Siad startup behavior.  The core functionality of the loading screen has been changed from being split across electron's `main` and `renderer` processes and now solely exists inside of the `renderer` process.  The IPC polling code used previously has been replaced with async callbacks and promises.  `detached` will also now be set on launch if `siad` is running before the UI is launched.

This should fix #216 as well as the bug mentioned [here](https://bitcointalk.org/index.php?topic=1060294.msg14768506#msg14768506).

### CODE REVIEW

File: js/rendererjs/loadingScreen.js [hunk 1]
```diff
@@ -1,64 +1,96 @@
...
+				if (error) {
+					showError(error);
+				} else {
+					startUI('Welcome to Sia', initUI);
+				}
+			});
+		});
+	}).catch(() => {
+		// config.path doesn't exist.  Prompt the user for siad's location
+		dialog.showErrorBox('Siad not found', 'Sia-UI couldnt locate siad.  Please navigate to siad.');
```
Comment (lukechampine) [inline js/rendererjs/loadingScreen.js]: couldn't

File: js/rendererjs/loadingScreen.js [hunk 2]
```diff
@@ -1,64 +1,96 @@
...
+		} else {
+			reject();
+		}
+	});
+});
+// startUI starts a Sia UI instance using the given welcome message.
+// calls initUI() after displaying a welcome message.
+const startUI = (welcomemsg, initUI) => {
+	// Display a welcome message, then initialize the ui
+	overlayText.innerHTML = welcomemsg;
```
Comment (lukechampine) [inline js/rendererjs/loadingScreen.js]: prefer camelCase, i.e. `welcomeMsg`

**Comment by lukechampine** [comment]
LGTM. So we have native support for arrow functions now?

**Comment by avahowell** [comment]
It would appear so.  Yay churn!

**Pull request merged**